src/marinade/infos.ts

```diff
@@ -0,0 +1,165 @@
+
+    // const testLayout = struct(
+    //   [
+    //     blob(8, "identifier"),
+    //     publicKey("msolMint"),
+    //     publicKey("adminAuthority"),
+    //     publicKey("operationalSolAccount"),
+    //     publicKey("treasuryMsolAccount"),
+    //     u8("reserveBumpSeed"),
+    //   ],
+    //   "State"
+    // );
+
+    const decodeData = MARINADE_FINANCE_ACCOUNT_STATE.decode(data);
+    console.log({ decodeData });
+    const {
+      msolMint,
+      adminAuthority,
+      operationalSolAccount,
+      treasuryMsolAccount,
+      reserveBumpSeed,
+      msolMintAuthorityBumpSeed,
+      rentExemptForTokenAcc,
+      rewardFee,
+      stakeSystem,
+      validatorSystem,
+      liqPool,
+      availableReserveBalance,
+      msolSupply,
+      msolPrice,
+      circulatingTicketCount,
+      circulatingTicketBalance,
+      lentFromReserve,
+      minDeposit,
+      minWithdraw,
+      stakingSolCap,
+      emergencyCoolingDown,
+    } = decodeData;
+
+    // Ensure the mint matches
+    if (!this._isAllowed(msolMint)) {
+      console.log(msolMint.toString());
+      throw Error("Error: Not a mSOL token account");
+    }
+
+    return {
+      vaultId,
+      msolMint,
+      adminAuthority,
+      operationalSolAccount,
+      treasuryMsolAccount,
+      reserveBumpSeed,
+      msolMintAuthorityBumpSeed,
+      rentExemptForTokenAcc,
+      rewardFee,
+      stakeSystem,
+      validatorSystem,
+      liqPool,
+      availableReserveBalance,
+      msolSupply,
+      msolPrice,
+      circulatingTicketCount,
+      circulatingTicketBalance,
+      lentFromReserve,
+      minDeposit,
+      minWithdraw,
+      stakingSolCap,
+      emergencyCoolingDown,
+    } as types.VaultInfo;
+  }
+
+  static async getAllDepositors(connection: Connection, userKey: PublicKey): Promise<types.DepositorInfo[]> {
+    const { value: accountInfos } = await connection.getTokenAccountsByOwner(userKey, {
+      mint: MSOL_MINT_ADDRESS,
+    });
+    const depositors = accountInfos.map((depositor) => this.parseDepositor(depositor.account.data, depositor.pubkey));
+    return depositors;
+  }
+
+  static async getDepositor(connection: Connection, depositorId: PublicKey): Promise<types.DepositorInfo> {
+    const depositorAccountInfo = await connection.getAccountInfo(depositorId);
+    if (!depositorAccountInfo) throw Error("Error: Cannot get depositor account");
+
+    return this.parseDepositor(depositorAccountInfo.data, depositorId);
+  }
+
+  static getDepositorId(
+    _vaultId: PublicKey,
+    userKey: PublicKey,
+    _programId: PublicKey = MARINADE_PROGRAM_ID
+  ): PublicKey {
+    return PublicKey.findProgramAddressSync(
+      [userKey.toBuffer(), TOKEN_PROGRAM_ID.toBuffer(), MSOL_MINT_ADDRESS.toBuffer()],
+      ASSOCIATED_TOKEN_PROGRAM_ID
+    )[0];
+  }
+
+  static parseDepositor(data: Buffer, depositorId: PublicKey): types.DepositorInfo {
+    const decodeData = MARINADE_FINANCE_ACCOUNT_STATE.decode(data);
```

Comment:
We are using Token account as depositor here.
The layout here is wrong

Reply:
Can you clarify?